### PR TITLE
fix: update form methods when username is updated

### LINKED ISSUE #5401: [CAL-326] Unable to update username
> From [SyncLinear.com](https://synclinear.com)

### PR DESCRIPTION
**Pull request by Udit-takkar**
Fixes: https://github.com/calcom/cal.com/issues/5401

Video:- 

https://user-images.githubusercontent.com/53316345/200375806-954ab8a8-08cb-44fe-a52f-cf253c39391c.mov

How to reproduce bug:-
https://threads.com/34432980734?s=ucqt4HpZQcM3tsQmy5m8G5

### CODE REVIEW

File: apps/web/pages/settings/my-account/profile.tsx [hunk 1]
```diff
@@ -69,7 +69,10 @@ const ProfileView = () => {
   const [currentUsername, setCurrentUsername] = useState<string | undefined>(user?.username || undefined);
```
Comment (emrysal) [inline apps/web/pages/settings/my-account/profile.tsx:69]:
```suggestion
  const [currentUsername, setCurrentUsername] = useState<string>();
```

File: apps/web/pages/settings/my-account/profile.tsx [hunk 2]
```diff
@@ -69,7 +69,10 @@ const ProfileView = () => {
...
   const [inputUsernameValue, setInputUsernameValue] = useState(currentUsername);
```
Comment (emrysal) [inline apps/web/pages/settings/my-account/profile.tsx:70]: This looks to me dependent on currentUsername could be broken

**Review by emrysal — CHANGES_REQUESTED** [review]

**Comment by emrysal** [comment]
Opened a PR against this branch @ https://github.com/calcom/cal.com/pull/5405 with a couple of improvements RHF wise.

**Comment by Udit-takkar** [comment]
@emrysal  there are few issues in the PR after you merged.

- UserSettings component in /gettting-started page also use UsernameAvailability component and on that component we are using user data from getServerSideProps instead of trpc.useQuery(["viewer.me"])  so currentUsername props does not get updated when you save username.

- i can't really fix this refCallback type error. if i change type of RefCallback i cannot access .current prop.

**Comment by Udit-takkar** [comment]
@emrysal  everything working fine now 👍

**Review by emrysal — APPROVED** [review]

**Pull request merged**